src/client/src/views/CaptionForm.js

```diff
@@ -1,11 +1,43 @@
 // This is the caption editor page, for filling in the details of a particular caption string
 
 const m = require('mithril')
+const Scene = require('../models/Scene')
 
 module.exports = {
+  oninit: function(vnode) {
+    // first set the current Scene, then set the current caption.
+    Scene.setCurrentScene(vnode.attrs.sceneId - 1)
+    Scene.setCurrentCaption(vnode.attrs.captionId - 1)
+  },
   view: function(vnode) {
     return m('', [
-      m('h1', `Caption ${vnode.attrs.id}`),
+      m('h1', Scene.currentCaption.name ? Scene.currentCaption.name : `Caption ${vnode.attrs.captionId}`),
```

Comment:
This is the line that was causing the issues. Change 
`Caption ${vnode.attrs.captionId}` 
to 
`Caption ${Scene.currentCaption.id}`